FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,20 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		self.client.get(from: url) { [weak self] clientResult in
+
+			guard self != nil else { return }
+
+			switch clientResult {
+			case .success((let data, let response)):
+				if let feedImages = try? FeedImageMapper.map(data, response) {
+					completion(.success(feedImages))
+				} else {
+					completion(.failure(Error.invalidData))
+				}
```

Comment:
Following from my previous comment, by having the `map` function `throw` the correct error we don't have to overwrite it here. That way we could use a `do/catch` statement and complete with the `error` caught.

Or we could use `Result`'s [initializer](https://developer.apple.com/documentation/swift/result/3139399-init) that takes a throwing closure as parameter and returns `.success` with the value returned, or `.failure` with any thrown error:
```suggestion
				completion(Result { try FeedImageMapper.map(data, response) })
```

Reply:
@thiagopenna The Result init point is very useful, Thanks